### PR TITLE
InventoryIterator

### PR DESCRIPTION
**Pull request by GetLocalPlayer**
Not sure for stdlib, but this is a quite common task for me.

### CODE REVIEW

**Comment by Frotty** [comment]
Good in theory, but I'm not a big fan of `unit.iterator()` for items

**Comment by Frotty** [comment]
Good in theory, but I'm not a big fan of `unit.iterator()` for items

**Comment by Frotty** [comment]
Good in theory, but I'm not a big fan of `unit.iterator()` for items

**Comment by GetLocalPlayer** [comment]
What if unit.inventory? I mean
```
for i in anyUnit.inventory()
    // ...
```

**Comment by Frotty** [comment]
Yea i think that's better if it works. You should also add some hotdoc with example code to the .inventory function then.

**Comment by Frotty** [comment]
Status?

**Comment by GetLocalPlayer** [comment]
Done. Added functions from previous PRs (should I do it? not sure).

File: wurst/_handles/Unit.wurst [hunk 1]
```diff
@@ -612,3 +633,35 @@ public function unit.incAbilityLevel(int abilityId)
 
 public function unit.decAbilityLevel(int abilityId)
 	DecUnitAbilityLevel(this, abilityId)
+	
+/* Iterator */
```
Comment (Frotty) [inline wurst/_handles/Unit.wurst]: I think regarding comments/doc, either write something useful, or nothing at all. This one doesn't really seem useful

File: wurst/_handles/Unit.wurst [hunk 2]
```diff
@@ -612,3 +633,35 @@ public function unit.incAbilityLevel(int abilityId)
...
+
+class InventoryIterator
+	private item array[6] items
+	private int count = 0
+
+	construct(unit u)
+		for slot = UnitInventorySize(u)-1 downto 0
```
Comment (Frotty) [inline wurst/_handles/Unit.wurst]: spacing around the minus operator

**Comment by GetLocalPlayer** [comment]
Done.

**Comment by Frotty** [comment]
Thanks 🍬

**Pull request merged**